src/lustre/lustreAstHelpers.ml

```diff
@@ -1086,15 +1116,18 @@ let rec abstract_pre_subexpressions: expr -> expr = function
   | Call (p, i, es) -> Call (p, i, List.map abstract_pre_subexpressions es) 
   | CallParam (p, i, tys, es) -> CallParam (p, i, tys, List.map abstract_pre_subexpressions es) 
                                    
-
-
-let extract_equation: node_item list -> node_equation list
-  = let extract_equation_helper: node_item -> node_equation list
-      = function
-      | Body n -> [n]
-      | _ -> []
-    in fun items ->
-       List.concat (List.map extract_equation_helper items)
+let extract_equation: node_item list -> node_equation list =
+  let rec extract_equation_helper ni = (match ni with
+    | Body eqn -> [eqn]
+    | IfBlock (_, _, l1, l2) -> 
+      List.flatten (List.map extract_equation_helper l1) @ 
+      List.flatten (List.map extract_equation_helper l2)
+    | FrameBlock (_, _, nes, nis) -> 
+      nes @ List.flatten (List.map extract_equation_helper nis)
+    | AnnotMain _ -> []
+    | AnnotProperty _ -> [])
+  in 
+  fun items -> List.flatten (List.map extract_equation_helper items) 
```

Comment:
This function is never called. I have my doubts about the usefulness of this function in future code since we are extracting equations without taking into consideration `IfBlocks` and `FrameBlock`. I would remove it.